### PR TITLE
feat(mobile): balance wheel animated graph bb-196

### PR DESCRIPTION
**Pull request by Shcracoziabra**
### Purpose
Implement configurable Wheel animated graph 
### Overview
The implemented Wheel component uses children components LabelOnCircle and AnimatedSector, which use react-native-svg and react-native-reanimated libraries.

**LabelOnCircle component:**

- uses Text and Circle component from react-native-svg,
- calculates elements positioning with custom helpers.

**AnimatedSector component:**

- wraps Circle component configured with r,stroke, dashArray, dashGap and dashOffset properties to look like a sector,
- uses Anumated.createAnimatedComponent and withTiming  from react-native-reanimated to implement wheel rising from center animation.

**Custom helpers created:**

- to calculate point coordinates on laying on circumference, using circle radius and percent of the circumference starting from top,
- to calculate radians from percent of the circumference starting from top

### Screenshots ###
balance wheel graph

### CODE REVIEW

File: apps/mobile/src/libs/components/wheel/libs/components/label-on-circle/label-on-circle.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			</SvgText>
+			<SvgText
+				dx={textX}
+				dy={textY}
+				fill="#000"
+				fontFamily="Nunito-Bold"
+				fontSize={wheelRadius * LabelCoefficient.SCORE_SIZE}
+				textAnchor="middle"
+				x={0}
+				y={0}
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/wheel/libs/components/label-on-circle/label-on-circle.tsx:69]: Avoid magic numbers

File: apps/mobile/src/libs/components/wheel/libs/helpers/get-sector-parameters/get-sector-parameters.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+type GetSectorParametersArguments = {
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/wheel/libs/helpers/get-sector-parameters/get-sector-parameters.ts:1]: Change file name to `get-sector-parameters.helper.ts`

File: apps/mobile/src/libs/components/wheel/libs/helpers/get-wheel-category-parameters/get-wheel-category-parameters.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { type ValueOf } from "shared";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/wheel/libs/helpers/get-wheel-category-parameters/get-wheel-category-parameters.ts:1]: Shouldn't import directly from shared

Comment (fshabanov) [inline apps/mobile/src/libs/components/wheel/libs/helpers/get-wheel-category-parameters/get-wheel-category-parameters.ts:1]: Change file name to `get-wheel-category-parameters.helper.ts`

File: apps/mobile/src/libs/types/wheel/wheel-data-item.type.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { type ValueOf } from "shared";
```
Comment (fshabanov) [inline apps/mobile/src/libs/types/wheel/wheel-data-item.type.ts:1]: Shouldn't import from shared

File: packages/shared/src/libs/helpers/get-circumference-point-coordinates/get-circumference-point-coordinates.helper.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+type GetPointCoordinatesArguments = {
```
Comment (fshabanov) [inline packages/shared/src/libs/helpers/get-circumference-point-coordinates/get-circumference-point-coordinates.helper.ts:1]: Not sure if it is a shared helper

File: packages/shared/src/libs/helpers/get-radians-from-percent/get-radians-from-percent.helper.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const getRadiansFromPercent = (percent: number): number => {
```
Comment (fshabanov) [inline packages/shared/src/libs/helpers/get-radians-from-percent/get-radians-from-percent.helper.ts:1]: Same, not sure if it should be in shared

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/quiz-entry/quiz-entry.tsx [hunk 7]
```diff
@@ -35,12 +49,13 @@ const QuizEntry: React.FC = () => {
 					<Text preset="heading" style={styles.text} weight="extraBold">
 						Craft your personal{"\n"}Life Balance Wheel!
 					</Text>
-					<Image
-						source={
-							require("~/assets/images/balance-wheel.png") as ImageSourcePropType
-						}
-						style={styles.image}
-					/>
+					<View style={globalStyles.alignItemsCenter}>
```
Comment (revenets) [inline apps/mobile/src/screens/quiz-entry/quiz-entry.tsx:52]: Just a nitpick, but let's leave here the image and use the actual wheel on the Wheel tab of bottom slider

File: apps/mobile/src/libs/components/wheel/libs/components/animated-sector/animated-sector.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+	const { dashArrayDash, dashArrayGap, dashOffset, radius, strokeWidth } =
+		getSectorParameters({
+			endPercent,
+			height,
+			holeSize,
+			layerOffset,
+			startPercent,
+		});
+
+	const ZERO = 0;
```
Comment (revenets) [inline apps/mobile/src/libs/components/wheel/libs/components/animated-sector/animated-sector.tsx:44]: Move this constant outside the component

File: apps/mobile/src/libs/components/wheel/wheel.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+};
+
+const Wheel: React.FC<Properties> = ({
+	categoriesData,
+	isLabelShown = true,
+	maxScore,
+	size,
+}: Properties) => {
+	const MINIFY_TWICE_COEFFICIENT = 0.5;
+	const ANIMATION_TIME = 500;
```
Comment (revenets) [inline apps/mobile/src/libs/components/wheel/wheel.tsx:26]: The same here - put constants outside

File: apps/mobile/src/libs/components/wheel/libs/helpers/get-wheel-category-parameters/get-wheel-category-parameters.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+	categoriesLength,
+	colors,
+	index,
+	maxScore,
+	score,
+	wheelRadius,
+}) => {
+	const MAX_PERCENT = 100;
+	const INNER_COLOR_OFFSET = 1;
+	const OUTER_COLOR_OFFSET = 0;
```
Comment (revenets) [inline apps/mobile/src/libs/components/wheel/libs/helpers/get-wheel-category-parameters/get-wheel-category-parameters.ts:36]:
```suggestion
const MAX_PERCENT = 100;
const INNER_COLOR_OFFSET = 1;
const OUTER_COLOR_OFFSET = 0;

const getWheelCategoryParameters: GetWheelCategoryParameters = ({
	categoriesLength,
	colors,
	index,
	maxScore,
	score,
	wheelRadius,
}) => {
```

File: apps/mobile/src/libs/components/wheel/wheel.tsx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+							wheelRadius={wheelRadius}
+						/>
+					)}
+
+					<AnimatedSector
+						animationTime={ANIMATION_TIME}
+						centerPoint={centerPoint}
+						endPercent={endPercent}
+						height={height}
+						holeSize={wheelHoleSize}
```
Comment (revenets) [inline apps/mobile/src/libs/components/wheel/wheel.tsx:97]: If we talk about the circle in the center, let's name this prop as `centerPadding` or `centerGap`.

**Review by revenets — CHANGES_REQUESTED** [review]

**Review by WilhelmWesser — APPROVED** [review]

File: apps/mobile/src/libs/components/wheel/wheel.tsx (hunk 9 above)
Comment (Shcracoziabra) [reply apps/mobile/src/libs/components/wheel/wheel.tsx:26]: May I put constants from all Wheel-related components into Wheel constants file? If so, is it allowed to list and export all those in wheel/libs/constants/constants.ts ?

Comment (revenets) [reply apps/mobile/src/libs/components/wheel/wheel.tsx:28]: You don't need to put them in a separate file. Just declare them in the same file but before the component (below the `type Properties...`)

File: apps/mobile/src/libs/components/wheel/libs/helpers/get-sector-parameters/get-sector-parameters.helper.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+}: GetSectorParametersArguments): {
+	dashArrayDash: number;
+	dashArrayGap: number;
+	dashOffset: number;
+	radius: number;
+	strokeWidth: number;
+} => {
+	const QUATER_PERCENT = 25;
+	const TWO = 2;
+	const MAX_PERCENT = 100;
```
Comment (revenets) [inline apps/mobile/src/libs/components/wheel/libs/helpers/get-sector-parameters/get-sector-parameters.helper.ts:24]: Please move these constants outside the helper as well.

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/wheel/wheel.tsx (hunk 9 above)
Comment (revenets) [reply apps/mobile/src/libs/components/wheel/wheel.tsx:28]: Also, please don't resolve conversation if you have some unclosed questions or didn't apply the changes.

**Review by revenets — APPROVED** [review]

File: apps/mobile/src/libs/components/wheel/libs/components/label-on-circle/label-on-circle.tsx [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+type Properties = {
+	centerPoint: number;
+	circleColor: string;
+	pointAngleRadians: number;
+	score: number;
+	text: string;
+	wheelRadius: number;
+};
+
+const ZERO = 0;
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/wheel/libs/components/label-on-circle/label-on-circle.tsx:17]:
```suggestion
const SVG_INITIAL_COORDINATE = 0;
```
Let's be more specific here

File: apps/mobile/src/libs/components/wheel/libs/components/animated-sector/animated-sector.tsx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+	centerGap: number;
+	centerPoint: number;
+	endPercent: number;
+	height: number;
+	layerOffset: number;
+	startPercent: number;
+	stroke: string;
+};
+
+const ZERO = 0;
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/wheel/libs/components/animated-sector/animated-sector.tsx:23]:
```suggestion
const INITIAL_ANIMATED_VALUE = 0;
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**